packages/@aws-cdk/core/lib/stack-synthesizers/stack-synthesizer.ts

```diff
@@ -27,13 +34,31 @@ export abstract class StackSynthesizer implements IStackSynthesizer {
    */
   public abstract addFileAsset(asset: FileAssetSource): FileAssetLocation;
 
+  /**
+   * Register an external File Asset
+   *
+   * Returns the parameters that can be used to refer to the asset inside the template.
+   */
+  public addExternalFileAsset(_: ExternalFileAssetSource): FileAssetLocation {
+    throw new Error('addExternalFileAsset not supported.');
```

Comment:
The error message is generated in an abstract class, so there is actually no context whatsoever to draw on (since the abstract class has no members or useful classes for context). I've left in the constructor name though, thanks for the addition!